plugins/Relay/QueryList.js

```diff
@@ -27,15 +27,34 @@ class QueryList extends React.Component {
     if (!this.props.queries.count()) {
       return <div style={styles.empty}>No Relay Queries logged</div>;
     }
-    var rows = this.props.queries.valueSeq().map((q, i) => (
-      <Query
-        data={q}
-        isSelected={q.get('id') === this.props.selectedQuery}
-        key={q.get('id')}
-        oddRow={(i % 2) === 1}
-        onSelect={() => this.props.selectQuery(q.get('id'))}
-      />
-    )).toArray();
+
+    const rows = [];
+    let odd = false;
+    let lastRequestNumber = -1;
+    this.props.queries.forEach(query => {
+      const requestNumber = query.get('requestNumber');
+      if (lastRequestNumber !== requestNumber) {
```

Comment:
Would it not be better to first partition `this.props.queries` by `requestNumber` before rendering them? (What if queries from different `requestNumber` values are interleaved?)